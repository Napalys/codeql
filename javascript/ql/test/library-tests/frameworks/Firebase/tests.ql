import javascript

query predicate firebaseRef(DataFlow::SourceNode ref) {
  ref = ModelOutput::getATypeNode("FirebaseDBRef").asSource()
}

query predicate firebaseSnapshot(DataFlow::SourceNode snap) {
  snap = ModelOutput::getATypeNode("Snapshot").asSource()
}

query predicate firebaseVal(DataFlow::SourceNode val) {
  val = ModelOutput::getASourceNode("remote").asSource()
}

query predicate requestInputAccess(Http::RequestInputAccess acc) { any() }

query predicate responseSendArgument(Http::ResponseSendArgument send) { any() }

query predicate routeHandler(Http::RouteHandler handler) { any() }
