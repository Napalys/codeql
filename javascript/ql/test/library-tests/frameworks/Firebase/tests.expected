firebaseRef
| src/import_assign.ts:4:3:4:17 | db.ref("hello") |
| src/import_named.ts:4:3:4:17 | db.ref("hello") |
| src/import_star.ts:4:3:4:17 | db.ref("hello") |
| tst.js:5:1:5:22 | fb.data ... ef('x') |
| tst.js:5:1:8:2 | fb.data ... eRef\\n}) |
| tst.js:5:38:5:38 | x |
| tst.js:7:3:7:7 | x.ref |
| tst.js:7:3:7:14 | x.ref.parent |
| tst.js:10:1:10:25 | admin.d ... ef('x') |
| tst.js:10:1:13:2 | admin.d ... eRef\\n}) |
| tst.js:10:41:10:41 | x |
| tst.js:12:3:12:7 | x.ref |
| tst.js:12:3:12:14 | x.ref.parent |
| tst.js:15:1:15:27 | functio ... ef('x') |
| tst.js:15:38:15:38 | x |
| tst.js:17:3:17:7 | x.ref |
| tst.js:17:3:17:14 | x.ref.parent |
| tst.js:20:1:20:27 | functio ... ef('x') |
| tst.js:20:38:20:38 | x |
| tst.js:21:3:21:10 | x.before |
| tst.js:22:3:22:9 | x.after |
| tst.js:23:3:23:7 | x.ref |
| tst.js:23:3:23:14 | x.ref.parent |
| tst.js:32:12:32:42 | this.fi ... .ref(x) |
| tst.js:46:12:46:42 | this.fi ... .ref(x) |
| tst.js:50:12:50:34 | this.ge ... hild(x) |
| tst.js:50:12:50:48 | this.ge ... value') |
| tst.js:54:5:54:37 | this.fi ... ef('x') |
firebaseSnapshot
| tst.js:5:1:8:2 | fb.data ... eRef\\n}) |
| tst.js:5:38:5:38 | x |
| tst.js:10:1:13:2 | admin.d ... eRef\\n}) |
| tst.js:10:41:10:41 | x |
| tst.js:15:38:15:38 | x |
| tst.js:20:38:20:38 | x |
| tst.js:21:3:21:10 | x.before |
| tst.js:22:3:22:9 | x.after |
| tst.js:50:12:50:34 | this.ge ... hild(x) |
| tst.js:50:12:50:48 | this.ge ... value') |
| tst.js:61:36:61:36 | x |
firebaseVal
| tst.js:6:3:6:9 | x.val() |
| tst.js:11:3:11:9 | x.val() |
| tst.js:16:3:16:9 | x.val() |
| tst.js:21:3:21:16 | x.before.val() |
| tst.js:22:3:22:15 | x.after.val() |
| tst.js:61:36:61:36 | x |
requestInputAccess
| tst.js:72:52:72:65 | req.params.foo |
responseSendArgument
| tst.js:72:52:72:65 | req.params.foo |
routeHandler
| tst.js:72:27:72:69 | (req, r ... foo); } |
